src/crypto/algorithms/megolm.ts

```diff
@@ -592,7 +592,7 @@ class MegolmEncryption extends EncryptionAlgorithm {
         payload: IPayload,
     ): Promise<void> {
         const contentMap: Record<string, Record<string, IEncryptedContent>> = {};
-        const deviceInfoByDeviceId = new Map<string, DeviceInfo>();
+        const deviceInfoByDeviceId = new Map<string, Map<string, DeviceInfo>>();
```

Comment:
This seems more accurate now:

```suggestion
        const deviceInfoByUserIdAndDeviceId = new Map<string, Map<string, DeviceInfo>>();
```